core/src/main/resources/hudson/model/Job/buildTimeTrend_resources.js

```diff
@@ -33,3 +32,63 @@ function buildTimeTrend_displayBuilds(data) {
 	}
 	ts_refresh(p);
 }
+
+/**
+ * Generate SVG Icon
+ */
+function generateSVGIcon(iconName) {
+
+	const imagesURL = document.head.getAttribute('data-imagesurl');
+
+	const isInProgress = iconName.endsWith("anime");
+	let buildStatus = 'never-built';
+	switch (iconName) {
+		case 'red':
+		case 'red-anime':
+			buildStatus = 'last-failed';
+			break;
+		case 'yellow':
+		case 'yellow-anime':
+			buildStatus = 'last-unstable';
+			break;
+		case 'blue':
+		case 'blue-anime':
+			buildStatus = 'last-successful';
+			break;
+		case 'grey':
+		case 'grey-anime':
+		case 'disabled':
+		case 'disabled-anime':
+			buildStatus = 'last-disabled';
+			break;
+		case 'aborted':
+		case 'aborted-anime':
+			buildStatus = 'last-aborted';
+			break;
+		case 'nobuilt':
+		case 'nobuilt-anime':
+			buildStatus = 'never-built';
+			break
+	}
+
+	const svg1 = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+	svg1.setAttribute('class', 'svg-icon');
+	svg1.setAttribute('viewBox', "0 0 24 24");
+	const use1 = document.createElementNS('http://www.w3.org/2000/svg', 'use');
+	use1.setAttribute('href', imagesURL + '/build-status/build-status-sprite.svg#build-status-' + (isInProgress ? 'in-progress' : 'static'))
+	svg1.appendChild(use1);
+
+	const svg2 = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+	svg2.setAttribute('class', 'svg-icon icon-' + iconName + ' icon-sm');
+	svg2.setAttribute('viewBox', "0 0 24 24");
+	const use2 = document.createElementNS('http://www.w3.org/2000/svg', 'use');
+	use2.setAttribute('href', imagesURL + '/build-status/build-status-sprite.svg#' + buildStatus)
+	svg2.appendChild(use2);
+
+	const span = new Element('span', {class: 'build-status-icon__wrapper icon-' + iconName + ' icon-sm', style: 'width: 16px; height: 16px;'}).
```

Comment:
You shouldn’t need to set icon-sm and the style afaik, the css class should handle it?

Reply:
You're right. Thanks for the suggestion!